### PR TITLE
DOP-1883: clean up redirects for SEO

### PR DESCRIPTION
**Pull request by schmalliso**
## Pull Request Info

*This addresses redirection chains in the docs-ecosystem repo as part of the SEO project*

- Organizes redirects and removes redirect chains (organization was required to *find* the redirection chains)
- Ensures all redirects to docs.mongodb.com/ pages point at the canonical (with `/`) URL to avoid extra redirect steps

### Issue JIRA link:
https://jira.mongodb.org/browse/DOP-1883

Generated redirects file: [drivers-redirects.txt](https://github.com/mongodb/docs-ecosystem/files/6103518/drivers-redirects.txt)

### CODE REVIEW

File: config/redirects [hunk 1]
```diff
@@ -4,108 +4,142 @@ define: manual https://docs.mongodb.com/manual
...
+raw: drivers/use-cases/sensitive-data-encryption ->  ${base}/security/client-side-field-level-encryption-guide/
+
+
+###
+### Redirects for Legacy Drivers Content
+###
+raw: drivers/tutorials -> ${base}/
+raw: drivers/perl-internals -> ${base}/drivers/community-supported-drivers/
+raw: drivers/downloads -> ${base}/drivers/
+raw: drivers/syntax-table -> ${base}/drivers/
```
Comment (terakilobyte) [inline config/redirects:20]:
```suggestion
raw: drivers/perl-internals -> ${base}/community-supported-drivers/
raw: drivers/downloads -> ${base}/
raw: drivers/syntax-table -> ${base}/
```

File: config/redirects [hunk 2]
```diff
@@ -4,108 +4,142 @@ define: manual https://docs.mongodb.com/manual
...
 raw: drivers/cpp-bson-helper-functions -> https://github.com/mongodb/mongo-cxx-driver/wiki/BSON%20Helper%20Functions
 raw: drivers/cpp-to-sql-to-mongo-shell -> https://github.com/mongodb/mongo-cxx-driver/wiki
-raw: drivers/tools/administration-interfaces -> ${base}/tools
-raw: drivers/tutorial/configure-worker-roles-in-azure -> ${base}/platforms/windows-azure
-raw: drivers/tutorial/deploy-mongodb-worker-roles-in-azure-version-1-6 -> ${base}/platforms/windows-azure
-raw: drivers/tutorial/deploy-mongodb-worker-roles-in-azure -> ${base}/platforms/windows-azure
-raw: drivers/tutorial/build-an-application-to-connect-to-mongodb-on-azure -> ${base}/platforms/windows-azure
-raw: drivers/tutorial/use-java-dbobject-to-perform-saves -> ${base}/drivers/java
+raw: drivers/tutorial/download-and-compile-cpp-driver-legacy -> https://github.com/mongodb/mongo-cxx-driver/wiki/Download-and-Compile-the-Legacy-Driver
+raw: drivers/tutorial/authenticate-with-cpp-driver -> https://mongodb.github.io/mongo-cxx-driver
```
Comment (terakilobyte) [inline config/redirects:29]:
```suggestion
raw: drivers/tutorial/authenticate-with-cpp-driver -> http://mongocxx.org/
```

File: config/redirects [hunk 3]
```diff
@@ -4,108 +4,142 @@ define: manual https://docs.mongodb.com/manual
...
+
+# csharp
+raw: drivers/tutorial/authenticate-with-csharp-driver -> ${base}/csharp/
+raw: drivers/csharp-community-projects -> ${base}/drivers/csharp/
```
Comment (terakilobyte) [inline config/redirects:33]:
```suggestion
raw: drivers/csharp-community-projects -> ${base}/csharp/
```

File: config/redirects [hunk 4]
```diff
@@ -4,108 +4,142 @@ define: manual https://docs.mongodb.com/manual
...
 raw: drivers/tutorial/use-aggregation-framework-with-java-driver -> http://mongodb.github.io/mongo-java-driver/3.0/driver/getting-started/quick-tour
-raw: drivers/csharp-community-projects -> ${base}/drivers/csharp
-raw: drivers/tutorial/authenticate-with-csharp-driver -> http://mongodb.github.io/mongo-csharp-driver/2.2/reference/driver/authentication
-raw: drivers/tutorial/getting-started-with-csharp-driver -> http://mongodb.github.io/mongo-csharp-driver/2.2/getting_started
-raw: drivers/tutorial/serialize-documents-with-the-csharp-driver -> http://mongodb.github.io/mongo-csharp-driver/2.2/getting_started
-raw: drivers/tutorial/use-csharp-driver -> http://mongodb.github.io/mongo-csharp-driver/2.2/getting_started
-raw: drivers/tutorial/use-linq-queries-with-csharp-driver -> http://mongodb.github.io/mongo-csharp-driver/2.2/getting_started
-raw: drivers/tutorial/getting-started-with-ruby-on-rails-3 -> ${base}/tutorial/ruby-bson-tutorial
-raw: drivers/tutorial/getting-started-with-ruby-on-rails -> ${base}/tutorial/ruby-bson-tutorial
+raw: drivers/tutorial/use-java-dbobject-to-perform-saves -> ${base}/drivers/java/
```
Comment (terakilobyte) [inline config/redirects:47]:
```suggestion
raw: drivers/tutorial/use-java-dbobject-to-perform-saves -> ${base}/java/
```

File: config/redirects [hunk 5]
```diff
@@ -4,108 +4,142 @@ define: manual https://docs.mongodb.com/manual
...
-raw: drivers/platforms/modulus -> https://docs.atlas.mongodb.com
-raw: drivers/platforms/rackspace-cloud -> https://docs.atlas.mongodb.com
+
+# nodejs
+raw: drivers/javascript -> ${base}/drivers/node-js/
+
+# ruby and mongoid
+raw: drivers/tutorial/getting-started-with-ruby-on-rails-3 -> ${base}/tutorial/ruby-bson-tutorial/
+raw: drivers/tutorial/getting-started-with-ruby-on-rails -> ${base}/tutorial/ruby-bson-tutorial/
+raw: drivers/tutorial/model-data-for-ruby-on-rails -> ${base}/drivers/ruby/
```
Comment (terakilobyte) [inline config/redirects:56]:
```suggestion
raw: drivers/tutorial/getting-started-with-ruby-on-rails-3 -> https://docs.mongodb.com/ruby-driver/current/tutorials/bson-v4/
raw: drivers/tutorial/getting-started-with-ruby-on-rails ->https://docs.mongodb.com/ruby-driver/current/tutorials/bson-v4/
raw: drivers/tutorial/model-data-for-ruby-on-rails -> ${base}/ruby/
```

**Review by terakilobyte — CHANGES_REQUESTED** [review]
Found some broken links in testing, and suggested direct links instead of links that follow a redirect in other cases.

**Comment by schmalliso** [comment]
@terakilobyte Good catches - thanks! I've merged your suggestions.

**Review by terakilobyte — APPROVED** [review]
lgtm

**Pull request merged**